### PR TITLE
Resolved Task #4

### PR DESCRIPTION
**Pull request by wladengine**

### CODE REVIEW

**Review by ASemykinBrl — CHANGES_REQUESTED** [review]
When I removed a semaphore from here the input doesn't seem to change. Can you update the impl to use it? Recursive calls should actually be synchronized

File: MultiThreading.Task4.Threads.Join/Program.cs [hunk 1]
```diff
@@ -10,11 +10,15 @@
  */
 
 using System;
+using System.Threading;
 
 namespace MultiThreading.Task4.Threads.Join;
 
 class Program
 {
+    private static readonly Semaphore _mainSemaphore = new(initialCount: 0, maximumCount: 1);
```
Comment (ASemykinBrl) [inline MultiThreading.Task4.Threads.Join/Program.cs:19]: You can remove this one easily, plays no role.

**Review by ASemykinBrl — CHANGES_REQUESTED** [review]
See comments

File: MultiThreading.Task4.Threads.Join/Program.cs (hunk 1 above)
Comment (wladengine) [reply MultiThreading.Task4.Threads.Join/Program.cs:19]: This is a synchronization semaphore of main function. If I remove all _mainSemaphore calls, I'll see "Done" in command line before all recursive calls will finish.

**Pull request merged**